compiler/qsc_formatter/src/main.rs

```diff
@@ -1,17 +1,195 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use std::env;
+use std::{
+    env,
+    fmt::Display,
+    path::{Path, PathBuf},
+};
 mod formatter;
-use formatter::format_str;
-
-fn main() {
-    let args: Vec<String> = env::args().collect();
-    let path = args.get(1).expect("expected path to file to format");
-    // read file from `path` into buffer
-    let file_as_string = std::fs::read_to_string(path).expect("file not found");
-    // format the buffer
-    let formatted = format_str(&file_as_string);
-    // write the formatted buffer back to `path`
-    std::fs::write(path, formatted).expect("could not write to file");
+use clap::{crate_version, Parser};
+use formatter::{calculate_format_edits, format_str};
+
+#[derive(Debug, Parser)]
+#[command(name = "fmt", version = crate_version!())]
+#[command(author, about, next_line_help = true)]
+struct Cli {
+    /// Path to the file or folder contain files to be formatted.
+    #[arg()]
+    path: PathBuf,
+
+    /// Search for Q# files recursively when a folder is given as PATH.
+    #[arg(short, long, default_value("false"))]
+    recursive: bool,
+
+    /// Overwrite the file contents with the formatted contents.
+    #[arg(short, long, default_value("false"))]
+    write: bool,
```

Comment:
Is "write" a common option in this case? If not, would you consider something a little more menacing, such as "overwrite"?

Reply:
`--write` or `-w` seems to be common for cli options where you want to specify that you are writing to a file. Prettier, for example, uses `--write`. I'm not sure I like `--overwrite` because the shorthand of that would be `-o` which I don't associate with writing. That's just a personal preference.